File: TP01/Q10/arquivo.c
#include <stdio.h>
int main(void)
{
    // VARIAVEIS
    int n = 0;
    double num = 0.0;
    FILE *arquivo = fopen("Q10.txt", "w");

    scanf("%d", &n);
    //  escrever N numeros no arquivo
    for (int i = 0; i < n; i++)
    {
        scanf("%lf", &num);
        fwrite(&num, sizeof(num), 1, arquivo);
    }
    fclose(arquivo);
    // abrir arquivo e ler N numeros de tras pra frente
    FILE *entrada = fopen("Q10.txt", "r");
    for (int i = 1; i <= n; i++)
    {
        fseek(entrada, -i * 8, SEEK_END);
        fread(&num, sizeof(double), 1, entrada);
        if (num - (int)num == 0)
        {
            printf("%d\n", (int)num);
        }
        else
        {
            printf("%g\n", num);
        }
    }
    fclose(entrada);
}

File: TP01/Q09/arquivo.java
import java.io.*;

public class arquivo {
    public static void main(String[] args) {
        // VARIAVEIS
        int n = MyIO.readInt();
        double num = 0.0;
        long fp = 0;
        // abrir arquivo e escrever n numeros no arquivo
        try {
            RandomAccessFile raf = new RandomAccessFile("Q09.txt", "rw");
            for(int i = 0; i < n; i++){
                num = MyIO.readDouble();
                raf.writeDouble(num);
                fp = raf.getFilePointer();
            }
            raf.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        // abrir novamente o arquivo, ler numeros de tras pra frente e imprimir
        try {
            RandomAccessFile raf = new RandomAccessFile("Q09.txt", "r");
            fp = fp - 8;
            for(int i = 0; i < n; i++){
                raf.seek(fp - i * 8);
                num = raf.readDouble();
                if ((num % 1) == 0){
                    MyIO.println((int)num);
                }
                else {
                    MyIO.println(num);
                }
            }
            raf.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}
